controllers/managedocs_controller.go

```diff
@@ -1057,6 +1067,40 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 			},
 			Key: alertRelabelConfigSecretKey,
 		}
+		if r.DeploymentType != convergedDeploymentType {
+			if err := r.get(r.rhobsRemoteWriteConfigSecret); err == nil {
+				rhobsSecretData := r.rhobsRemoteWriteConfigSecret.Data
+				if _, found := rhobsSecretData[rhobsRemoteWriteConfigIdSecretKey]; !found {
+					return fmt.Errorf("rhobs secret does not contain a value for key %v", rhobsRemoteWriteConfigIdSecretKey)
+				}
+
+				if _, found := rhobsSecretData[rhobsRemoteWriteConfigSecretName]; !found {
+					return fmt.Errorf("rhobs secret does not contain a value for key %v", rhobsRemoteWriteConfigSecretName)
+				}
+
+				rhobsAudience, found := rhobsSecretData["rhobs-audience"]
+				if !found {
+					return fmt.Errorf("rhobs secret does not contain a value for key rhobs-audience")
+				}
+				remoteWriteSpec := &r.prometheus.Spec.RemoteWrite[0]
+				remoteWriteSpec.URL = r.RHOBSEndpoint
+				remoteWriteSpec.OAuth2.ClientID.Secret.LocalObjectReference.Name = r.RHOBSSecretName
+				remoteWriteSpec.OAuth2.ClientID.Secret.Key = rhobsRemoteWriteConfigIdSecretKey
+				remoteWriteSpec.OAuth2.ClientSecret.LocalObjectReference.Name = r.RHOBSSecretName
+				remoteWriteSpec.OAuth2.ClientSecret.Key = rhobsRemoteWriteConfigSecretName
+				remoteWriteSpec.OAuth2.TokenURL = r.RHSSOTokenEndpoint
+				remoteWriteSpec.OAuth2.EndpointParams["audience"] = string(rhobsAudience)
+			} else {
+				if errors.IsNotFound(err) {
+					r.Log.Info("RHOBS remote write config secret not found , disabling remote write")
+					r.prometheus.Spec.RemoteWrite = nil
+				} else {
+					return err
+				}
+			}
```

Comment:
We have a common pattern for handling these kind of cases, we handle error first then have an `if` to select between the non-error paths

```suggestion
			if err := r.get(r.rhobsRemoteWriteConfigSecret); err != nil &&  !errors.IsNotFound(err) {
				return err
			}
			
			if r.rhobsRemoteWriteConfigSecret.UID != "" {
				rhobsSecretData := r.rhobsRemoteWriteConfigSecret.Data
				if _, found := rhobsSecretData[rhobsRemoteWriteConfigIdSecretKey]; !found {
					return fmt.Errorf("rhobs secret does not contain a value for key %v", rhobsRemoteWriteConfigIdSecretKey)
				}

				if _, found := rhobsSecretData[rhobsRemoteWriteConfigSecretName]; !found {
					return fmt.Errorf("rhobs secret does not contain a value for key %v", rhobsRemoteWriteConfigSecretName)
				}

				rhobsAudience, found := rhobsSecretData["rhobs-audience"]
				if !found {
					return fmt.Errorf("rhobs secret does not contain a value for key rhobs-audience")
				}
				remoteWriteSpec := &r.prometheus.Spec.RemoteWrite[0]
				remoteWriteSpec.URL = r.RHOBSEndpoint
				remoteWriteSpec.OAuth2.ClientID.Secret.LocalObjectReference.Name = r.RHOBSSecretName
				remoteWriteSpec.OAuth2.ClientID.Secret.Key = rhobsRemoteWriteConfigIdSecretKey
				remoteWriteSpec.OAuth2.ClientSecret.LocalObjectReference.Name = r.RHOBSSecretName
				remoteWriteSpec.OAuth2.ClientSecret.Key = rhobsRemoteWriteConfigSecretName
				remoteWriteSpec.OAuth2.TokenURL = r.RHSSOTokenEndpoint
				remoteWriteSpec.OAuth2.EndpointParams["audience"] = string(rhobsAudience)
			} else {
				r.Log.Info("RHOBS remote write config secret not found , disabling remote write")
				r.prometheus.Spec.RemoteWrite = nil
			}
```